### PR TITLE
Handle dbt state in fal flow run

### PR DESCRIPTION
**Pull request by mkam039**
Makes a state selector work with passed state.

```
fal flow run --select state:new --state my_state_dir --threads 1
```

### CODE REVIEW

File: integration_tests/features/steps/fal_steps.py [hunk 1]
```diff
@@ -36,15 +36,45 @@ def seed_data(context):
...
 
+@then("the following command will fail")
+def invoke_failing_fal_flow(context):
+    profiles_dir = Path(context.base_dir).parent.absolute()
+    args = context.text.replace("$baseDir", context.base_dir)
+    args = args.replace("$profilesDir", str(profiles_dir))
+    args = args.replace("$tempDir", str(context.temp_dir.name))
+    try:
+        cli(args.split(" "))
+        assert True is False
```
Comment (chamini2) [inline integration_tests/features/steps/fal_steps.py:68]:
```suggestion
        assert False, "Should have raised an exception"
```

File: integration_tests/features/steps/fal_steps.py [hunk 2]
```diff
@@ -36,15 +36,45 @@ def seed_data(context):
...
+    except Exception as e:
+        print(e)
+
+
+@then("model named {model_name} is removed")
+def delete_model(context, model_name):
+    os.system(f"rm {context.base_dir}/models/{model_name}.sql")
```
Comment (chamini2) [inline integration_tests/features/steps/fal_steps.py:75]: I think this will affect other tests, right? We have a temp dir for results, we do not copy the whole directory.

File: integration_tests/projects/001_flow_run_with_selectors/data/time_series.csv [hunk 3]
```diff
@@ -1,4 +1,4 @@
-id,y,ds
+,y,ds
```
Comment (chamini2) [inline integration_tests/projects/001_flow_run_with_selectors/data/time_series.csv:1]: Why this change?

**Review by chamini2 — CHANGES_REQUESTED** [review]

File: integration_tests/projects/001_flow_run_with_selectors/data/time_series.csv (hunk 3 above)
Comment (mkam039) [reply integration_tests/projects/001_flow_run_with_selectors/data/time_series.csv]: Already been merged, there was a conflict

File: integration_tests/features/steps/fal_steps.py (hunk 2 above)
Comment (mkam039) [reply integration_tests/features/steps/fal_steps.py:74]: Yes, but this is needed for updating models for `--state` tests. Any model update will mess with other tests. We have to be careful when using this step.

File: src/faldbt/project.py [hunk 4]
```diff
@@ -186,6 +186,7 @@ class FalDbt:
     models: List[DbtModel]
     tests: List[DbtTest]
     el: FalElClient
+    state: str
```
[the remark this answers is not in the document]
Comment (mkam039) [reply src/faldbt/project.py:189]: Done

File: integration_tests/features/steps/fal_steps.py (hunk 1 above)
Comment (mkam039) [reply integration_tests/features/steps/fal_steps.py]: Done

File: integration_tests/features/steps/fal_steps.py (hunk 2 above)
Comment (mkam039) [reply integration_tests/features/steps/fal_steps.py:74]: Great

File: src/faldbt/project.py [hunk 5]
```diff
@@ -186,6 +186,7 @@ class FalDbt:
     models: List[DbtModel]
     tests: List[DbtTest]
     el: FalElClient
+    state: Optional[str]
```
[the remark this answers is not in the document]
Comment (mkam039) [reply src/faldbt/project.py]: Good catch 😉

**Review by chamini2 — APPROVED** [review]
